From 7c854566c8dbbf047a90139a928a6ac9196546af Mon Sep 17 00:00:00 2001
Message-ID: <7c854566c8dbbf047a90139a928a6ac9196546af.1692093230.git.amy@centricular.com>
From: James Zern <jzern@google.com>
Date: Wed, 1 Feb 2023 13:27:06 -0800
Subject: [PATCH 2/3] vp9_diamond_search_sad_neon: use DECLARE_ALIGNED

rather than the gcc specific __attribute__((aligned())); fixes build
targeting ARM64 windows.

Bug: webm:1788
Change-Id: I2210fc215f44d90c1ce9dee9b54888eb1b78c99e
(cherry picked from commit 858a8c611f4c965078485860a6820e2135e6611b)
---
 vp9/encoder/arm/neon/vp9_diamond_search_sad_neon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vp9/encoder/arm/neon/vp9_diamond_search_sad_neon.c b/vp9/encoder/arm/neon/vp9_diamond_search_sad_neon.c
index 33753f77b..997775a66 100644
--- a/vp9/encoder/arm/neon/vp9_diamond_search_sad_neon.c
+++ b/vp9/encoder/arm/neon/vp9_diamond_search_sad_neon.c
@@ -220,7 +220,7 @@ int vp9_diamond_search_sad_neon(const MACROBLOCK *x,
       // Look up the component cost of the residual motion vector
       {
         uint32_t cost[4];
-        int16_t __attribute__((aligned(16))) rowcol[8];
+        DECLARE_ALIGNED(16, int16_t, rowcol[8]);
         vst1q_s16(rowcol, v_diff_mv_w);
 
         // Note: This is a use case for gather instruction
-- 
2.41.0

